library/src/test/java/org/apache/apex/malhar/lib/wal/FileSystemWALTest.java

```diff
@@ -53,20 +55,33 @@ private static Slice getRandomSlice(int len)
   {
     private String targetDir;
     FileSystemWAL fsWAL = new FileSystemWAL();
+    Configuration conf = new Configuration();
+    FileSystem fs;
 
     @Override
     protected void starting(Description description)
     {
-      TestUtils.deleteTargetTestClassFolder(description);
       targetDir = "target/" + description.getClassName() + "/" + description.getMethodName();
+
+      try {
```

Comment:
Yes, that is the motivation for these changes.  The current implementation of the WAL has different code path for a local file system so this issue was not caught for a long time. We need a way to run these tests on HDFS or Mock FS. As a first step, unit tests should only be using Hadoop's FileSystem api. I am planning to propose this in the mailing list.